src/golang/lib/workflow/validate.go

```diff
@@ -7,24 +7,34 @@ import (
 	"github.com/aqueducthq/aqueduct/lib/collections/shared"
 	"github.com/aqueducthq/aqueduct/lib/collections/workflow"
 	"github.com/aqueducthq/aqueduct/lib/database"
+	"github.com/aqueducthq/aqueduct/lib/graph"
+	"github.com/aqueducthq/aqueduct/lib/models"
 	"github.com/aqueducthq/aqueduct/lib/repos"
 	"github.com/aqueducthq/aqueduct/lib/workflow/utils"
 	"github.com/dropbox/godropbox/errors"
+	"github.com/google/uuid"
 )
 
 const (
 	internalValidationErrMsg = "Internal system error occurred while validating Workflow schedule."
 )
 
-// ValidateSchedule validates the provided schedule. The following conditions are
-// not allowed:
-// - Having a CascadingUpdateTrigger where SourceID is for a Workflow that is
+// ValidateSchedule validates the provided schedule. The flag isUpdate indicates
+// whether or not the Workflow, for which the schedule is being validated already exists.
+// If the Workflow exists, its ID should be provided as workflowID; otherwise,
+// workflowID can be ignored.
+// The following conditions are not allowed:
+// 1. Using a CascadingUpdateTrigger when engineType is Airflow.
+// 2. Having a CascadingUpdateTrigger where SourceID is for a Workflow that is
 // running on a non self-orchestrated engine, such as Airflow. This is not allowed
 // since Aqueduct cannot trigger Workflow runs at the end of execution on an
 // engine that is not self-orchestrated.
+// 3. Having a CascadingUpdateTrigger that creates a cycle amongst the cascading workflows.
 // It returns an HTTP status code and a client-friendly error, if any.
 func ValidateSchedule(
```

Comment:
just confirming that this function is called within the txn during workflow registration?

Reply:
Good catch, I moved it inside the txn